src/tox/tox_env/python/api.py

```diff
@@ -168,9 +168,14 @@ def _validate_base_python(
         if env_base_python is not None:
             spec_name = PythonSpec.from_string_spec(env_base_python)
             for base_python in base_pythons:
-                if Path(base_python).is_absolute():
-                    return [base_python]
                 spec_base = PythonSpec.from_string_spec(base_python)
+                if spec_base.path is not None:
+                    path = Path(spec_base.path).absolute()
+                    if str(spec_base.path) == sys.executable:
+                        ver, is_64 = sys.version_info, sys.maxsize != 2**32
```

Comment:
When reviewing https://github.com/fedora-python/tox-current-env/pull/78 @frenzymadness found out that `is_64` is always True:

### Fedora Rawhide i686

```pycon
>>> sys.maxsize
2147483647
>>> 2**32
4294967296
>>> sys.maxsize != 2**32
True
```

### Fedora Rawhide x86_64


```pycon
>>> sys.maxsize
9223372036854775807
>>> 2**32
4294967296
>>> sys.maxsize != 2**32
True
```

https://docs.python.org/3/library/platform.html#platform.architecture says:

> To get at the “64-bitness” of the current interpreter, it is more reliable to query the [sys.maxsize](https://docs.python.org/3/library/sys.html#sys.maxsize) attribute:
> 
> ```python
> is_64bits = sys.maxsize > 2**32
> ```

### Fedora Rawhide i686

```pycon
>>> sys.maxsize > 2**32
False
```

### Fedora Rawhide x86_64


```pycon
>>> sys.maxsize > 2**32
True
```

Reply:
Ah good catch both. can you put in a PR to fix it 😅

Reply:
I am working on it. I assume you intended a string like `cpython3.13-64`.

Reply:
That is correct, thank you.

Reply:
Drafted https://github.com/tox-dev/tox/pull/3327 -- waiting for the CI to validate it.